### ISSUE TITLE
Spyder crashes when rewriting a line of code

### ISSUE THREAD

**Issue by rotemgb** [spyder-ide/spyder#10710]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
When using auto-complete on a line of code, highlight that line and start writing, then spyder crashes.

![Peek 2019-11-14 09-56](https://user-images.githubusercontent.com/35784524/68883128-412ccb00-06c5-11ea-968c-56db9da11d2a.gif)

### What steps reproduce the problem?

1. Write a line of code and use auto-complete. For example, `for i in ran` and hit enter for auto-complete.
2. Highlight that line.
3. Start writing. 

### What is the expected output?
Spyder shouldn't crash.

### Paste Traceback/Error Below (if applicable)

```python-traceback

  File "/home/rotem/Projects/spyder/spyder/plugins/editor/extensions/snippets.py", line 228, in _on_key_pressed
    self._process_text(text)
  File "/home/rotem/Projects/spyder/spyder/plugins/editor/extensions/snippets.py", line 52, in wrapper
    return f(self, *args, **kwargs)
  File "/home/rotem/Projects/spyder/spyder/plugins/editor/extensions/snippets.py", line 236, in _process_text
    self.insert_text(text, line, column)
  File "/home/rotem/Projects/spyder/spyder/plugins/editor/extensions/snippets.py", line 368, in insert_text
    if node.name == 'EPSILON':
AttributeError: 'NoneType' object has no attribute 'name'


```

## Versions

* Spyder version: 4.0.0.dev0
* Python version: 3.7.5
* Qt version: 5.9.6
* PyQt version: 5.9.2
* Operating System name/version: Linux 5.3.0-20-generic, Pop_Os 19.10

### Dependencies

```
cloudpickle >=0.5.0          :  1.2.2 (OK)
pygments >=2.0               :  2.4.2 (OK)
qtconsole >=4.5.5            :  4.5.5 (OK)
nbconvert >=4.0              :  5.6.0 (OK)
sphinx >=0.6.6               :  2.2.1 (OK)
pylint >=0.25                :  2.4.3 (OK)
psutil >=0.3                 :  5.6.3 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
zmq >=17                     :  18.1.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
spyder_kernels >=1.7.0;<2.0.0:  1.7.0 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
watchdog                     :  None (OK)
keyring                      :  None (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pympler                      :  None (OK)
sympy >=0.7.3                :  None (NOK)
cython >=0.21                :  None (NOK)
IPython >=4.0                :  7.9.0 (OK)
matplotlib >=2.0.0           :  3.1.1 (OK)
pandas >=0.13.1              :  0.25.2 (OK)
numpy >=1.7                  :  1.17.3 (OK)
scipy >=0.17.0               :  None (NOK)
pyls >=0.29.3;<0.30.0        :  0.29.3 (OK)
xdg >=0.26                   :  0.26 (OK)
rtree >=0.8.3                :  0.8.3 (OK)
```

**Comment by goanpeca** [comment]
@steff456 is this solved by your PR ?

**Comment by steff456** [comment]
Yes! This should be fixed with PR #10688

**Comment by ccordoba12** [comment]
Thanks for reporting @rotemgb. This is a duplicate of issue #10650 and it'll be fixed in release candidate 2.
